### PR TITLE
Review book

### PR DESCRIPTION
**Pull request by dcthanh**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  before_action :correct_user!, only: [:edit, :update, :destroy]
+
+  before_filter :load_book
+
+  def new
+    @review = @book.reviews.build
+  end
+
+  def create
+    @review = @book.reviews.build(review_params)
```
Comment (chuanhtuan) [inline app/controllers/reviews_controller.rb]: ()

File: app/views/books/show.html.erb [hunk 2]
```diff
@@ -10,10 +10,24 @@
             <h1><%= @book.name %></h1><br>by <%= @book.author %><br>
             About the book :<br><h4><%= @book.desc %></h4><br>
             <%= link_to 'ADD LIBRARY', "#", class:"btn btn-warning" %>
+
+            <h3>Reviews of <em><%= @book.title %></em></h3>
+            <% if current_user %>
```
Comment (chuanhtuan) [inline app/views/books/show.html.erb]: <% if current_user? %>

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+class ReviewsController < ApplicationController
+  before_action :authenticate_user!, only: [:create, :edit, :update, :destroy]
```
Comment (dieunb) [inline app/controllers/reviews_controller.rb]: use `except` instead of `only` for this case

File: app/views/books/show.html.erb [hunk 4]
```diff
@@ -10,10 +10,24 @@
             <h1><%= @book.name %></h1><br>by <%= @book.author %><br>
             About the book :<br><h4><%= @book.desc %></h4><br>
             <%= link_to 'ADD LIBRARY', "#", class:"btn btn-warning" %>
+
+            <h3>Reviews of <em><%= @book.title %></em></h3>
+            <% if !current_user.nil? %>
```
Comment (dieunb) [inline app/views/books/show.html.erb]: unless

File: config/routes.rb [hunk 5]
```diff
@@ -7,7 +7,9 @@
 
   devise_for :users
   resources :users
-  resources :books
+  resources :books do
+    resources :reviews, only: [:new, :create]
```
Comment (dieunb) [inline config/routes.rb]: I see more actions at reviews_controller, but why do you define only new and create?

**Pull request merged**